src/main/java/at/hannibal2/skyhanni/features/inventory/ItemPickupLog.kt

```diff
@@ -0,0 +1,314 @@
+                return
+            }
+            if (isBannedItem(item)) return
+            itemsAddedToInventory[hash] = itemInfo
+        }
+    }
+
+    private data class UpdatedItem(val name: String, var amount: Long, val neuInternalName: NEUInternalName?) {
+        var time = SimpleTimeMark.now()
+
+        fun updateAmount(change: Long) {
+            amount += change
+            time = SimpleTimeMark.now()
+        }
+
+        fun isExpired() = time.passedSince() > config.expireAfter.seconds
+    }
+
+    private fun isEnabled() = LorenzUtils.inSkyBlock && config.enabled
+
+    private fun renderList(prefix: String, amount: Long, name: String, itemIcon: NEUInternalName?): Renderable {
+
+
+        return Renderable.horizontalContainer(
+            buildList {
+                for (item in config.displayLayout) {
+                    when (item) {
+                        DisplayLayout.ICON -> {
+                            if (itemIcon != null) {
+                                addItemStack(itemIcon)
+                            } else {
+                                ItemNameResolver.getInternalNameOrNull(name)?.let { addItemStack(it) }
+                            }
+                        }
+
+                        DisplayLayout.CHANGE_AMOUNT -> {
+                            val formattedAmount = if (config.shorten) amount.shortFormat() else amount.addSeparators()
+                            add(Renderable.string("${prefix}${formattedAmount}"))
+                        }
+
+                        DisplayLayout.ITEM_NAME -> {
+                            add(Renderable.string(name))
+                        }
+
+                        null -> {}
+                    }
+                }
+            },
+        )
+    }
+
+
+    private fun worldChangeCooldown(): Boolean {
+        return LorenzUtils.lastWorldSwitch.passedSince() > 2.seconds
+    }
+
+    private fun updateDisplay() {
+        if (!isEnabled()) return
+
+        val display = mutableListOf<Renderable>()
+
+        itemsAddedToInventory.values.removeIf { it.isExpired() }
+        itemsRemovedFromInventory.values.removeIf { it.isExpired() }
+
+        val removedItemsToNoLongerShow = itemsRemovedFromInventory.toMutableMap()
+        val addedItemsToNoLongerShow = itemsAddedToInventory.toMutableMap()
+
+        if (config.compactLines) {
+            val iterator = addedItemsToNoLongerShow.iterator()
+            while (iterator.hasNext()) {
+                val item = iterator.next()
+
+                if (removedItemsToNoLongerShow.containsKey(item.key)) {
+                    val it = removedItemsToNoLongerShow[item.key]
+                    if (it != null) {
+                        val currentTotalValue = item.value.amount - it.amount
+
+                        if (currentTotalValue > 0) {
+                            display.add(renderList("§a+", currentTotalValue, item.value.name, item.value.neuInternalName))
```

Comment:
create a utils function using  `renderList` and `item.value`, to avoid the many code duplications